src/Model/ModelData.php

```diff
@@ -505,87 +430,46 @@ protected function objCacheClear()
      *
      * @return object|DBField|null The specific object representing the field, or null if there is no
      * property, method, or dynamic data available for that field.
-     * Note that if there is a property or method that returns null, a relevant DBField instance will
-     * be returned.
      */
     public function obj(
```

Comment:
The main changes in this method are:
1. Remove unnecessary `cacheName` parameter
2. Cache the raw value, not the cast object (necessary so we are using the same cached values as `ViewLayerData`)
3. Rely on the new casting service for the actual casting
4. If there are arguments and a getter method, pass the arguments to the getter method.